org.eclipse.m2e.core/src/org/eclipse/m2e/core/project/IMavenProjectFacade.java

```diff
@@ -156,4 +156,18 @@ MojoExecution getMojoExecution(MojoExecutionKey mojoExecutionKey, IProgressMonit
   String getLifecycleMappingId();
 
   Map<MojoExecutionKey, List<IPluginExecutionMetadata>> getMojoExecutionMapping();
+
+  /**
+   * Iterates through the project's build plugins and finds any custom packaging types that are available
+   */
+  void updateAvailablePackagingTypes();
+
+  /**
+   * Gets any custom packaging types found using updateAvailablePackagingTypes. If this method is called while
+   * availablePackagingTypes is empty, updateAvailablePackagingTypes will be called
+   * 
+   * @see IMavenProjectFacade#updateAvailablePackagingTypes()
+   * @return A set of all custom packaging types available to this project, or an empty set if none are found
+   */
+  Set<String> getAvailablePackagingTypes();
```

Comment:
Since this is only useful for the UI (OverviewPage), I don't think we should make it an API here, and instead we should probably just move the code to OverviewPage directly.

Reply:
I think `getAvailablePackagingTypes()` is good here but  `updateAvailablePackagingTypes()` should not be part of the API.

Reply:
Why I think its useful in the API is that I would expect in the future that this information is read via the maven-extension mechanism and then we don't want to remember the Overviewpage uses some kind of own solution.
Also we probably should check elsewhere in the code if we have hard-coded defaults that better call this method instead.

Reply:
> Why I think its useful in the API is that I would expect in the future that this information is read via the maven-extension mechanism and then we don't want to remember the Overviewpage uses some kind of own solution.

That's not how API are supposed to be growing. Some code is promoted as an API if there is a clear need from it, from multiple consumers and if it makes sense to host the logic for those clients into the object because of some strong coupling. It's not the case here. IMavenProjectFacade represents one project and its info. The available types is most often useless for a project and not really related to the project but more to the classrealm, the set is moveable so it's unclear from API what to expect as value... It could be some utility class somewhere, but definitely not a getter in IMavenProjectFacade.
At this stage, there is no good reason to make it API, so we should avoid future pain and not make it API, at least not that way.
The more API != the better.

Reply:
Then lets move it to OverviewPage but include it in 1.20 ...

Reply:
Moved to overview page.